### PR TITLE
Issue #4801: add suppressions on the eleventh part of IDEA violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4801

add suppressions

### CODE REVIEW

File: config/intellij-idea-inspections.xml [hunk 1]
```diff
@@ -2167,16 +2168,17 @@
         <option name="ignoreSynchronizedSuperMethods" value="true" />
     </inspection_tool>
     <inspection_tool class="SynchronizedOnLiteralObject" enabled="true" level="WARNING" enabled_by_default="true" />
-    <inspection_tool class="SyntaxError" enabled="false" level="ERROR" enabled_by_default="false" />
+    <inspection_tool class="SyntaxError" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="SystemExit" enabled="false" level="ERROR" enabled_by_default="false" />
     <inspection_tool class="SystemGC" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="SystemGetenv" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="SystemOutErr" enabled="true" level="ERROR" enabled_by_default="true" />
+    <!-- we do not use system variables for any security relates reasons(no passwords or authentication etc. -->
```
Comment (rnveach) [inline config/intellij-idea-inspections.xml]: missing end `)`

File: config/intellij-idea-inspections.xml [hunk 2]
```diff
@@ -2275,6 +2279,8 @@
     <inspection_tool class="UNUSED_IMPORT" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="UnaryPlus" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="UncheckedExceptionClass" enabled="false" level="ERROR" enabled_by_default="false" />
+    <!-- raise a lot of violations that make no sense to benefit from parentheses,
+    like "a ==1 && b==2" force to use parentheses, that is not good -->
```
Comment (rnveach) [inline config/intellij-idea-inspections.xml]: extra space before `==`.

File: config/intellij-idea-inspections.xml (hunk 1 above)
Comment (Nimfadora) [inline config/intellij-idea-inspections.xml]: fixed

File: config/intellij-idea-inspections.xml (hunk 2 above)
Comment (Nimfadora) [inline config/intellij-idea-inspections.xml]: fixed

**Pull request merged**